MVVMIOS/ViewModel/MovieViewModel.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  MovieViewModel.swift
+//  MVVMIOS
+//
+//  Created by Ravi Bastola on 3/6/22.
+//
+
+import Foundation
+
+protocol MovieViewModelResultPresentable: AnyObject {
+    func presentResult(_ result: Result<[MovieResult], Error>?)
+}
+
+class MovieViewModel {
+
+    let service: NetworkingProtocol
+
+    weak var resultPresentable: MovieViewModelResultPresentable?
+
+    init(_ service: NetworkingProtocol) {
+        self.service = service
+    }
+
+    func fetchDataFromApi() {
+        guard let url = URL(string: "http://gdhistory.com/api/periods") else {
+            self.resultPresentable?.presentResult(.failure(NetworkError.invalidURL))
```

Comment:
As I mentioned above, don't just pass a raw network error here. A VC shouldn't know what to do in case of a network error - it shouldn't know a network is involved, or anything about the implementation of what its data came from. It should only know that it has new data to present. It doesn't know that data represents an error or a success, it just presents what it is told to present. 

Instead pass something like a `MovieViewState` of `.error(MovieErrorViewState)` (see above) that tells the VC exactly what to present to the user. The ViewModel can/should have method(s) for converting model state (such as errors or `[MovieResult]` into view state. Things like `internal func viewState(for error: Error) -> MovieViewState` and `internal func viewState(for movies: [MovieResult]) -> MovieViewState`. And each of these methods should have a unit tests to verify that they produce the correct view state for a given model state, including what if the `[MovieResult]` is empty? What if it only has one item? What if it has exactly 1 item, what if it has a few items? Similar for the error one. Does it properly handle the error state, etc. May seem like overkill, but that's why implementing a feature like you're working on would take a good engineer at least a full 40 hour week if not much longer.